lib/api.js

```diff
@@ -543,6 +550,83 @@ api.declare({
     return x;
   })});
 });
+
+api.declare({
+  method: 'delete',
+  route: '/credentials',
+  name: 'claimCredentials',
+  title: 'Request worker credentials',
+  stability: API.stability.experimental,
+  input: 'claim-credentials-request.json#',
+  description: [
+    'Yield credentials for the worker running in the given instance.',
```

Comment:
@walac and I had a quick chat about this over vidyo.  We decided that instead of doing any sort of scope protecting, we'd just resend the document and signature and use that to ensure the claimer is the right machine.  Since the trust is derived from these documents, they are as trustworthy as the taskcluster credentials.

This also makes worker implementation easier, it is a GET followed by a DELETE to the same API endpoint, which is much easier than a GET, parsing the document to build a DELETE URL and using TC credentials to make the request.